tests/test_atoms_state.py

```diff
@@ -0,0 +1,386 @@
+        self.add_slater_interactions(hubbard_interactions, slater_integrals)
+
+        # Resolving U, U', and J from class method
+        (
+            u_interaction,
+            u_interorbital_interaction,
+            j_hunds_coupling,
+        ) = hubbard_interactions.resolve_u_interactions(self.logger)
+
+        if None not in (u_interaction, u_interorbital_interaction, j_hunds_coupling):
+            assert np.isclose(u_interaction.to('eV').magnitude, results[0])
+            assert np.isclose(u_interorbital_interaction.to('eV').magnitude, results[1])
+            assert np.isclose(j_hunds_coupling.to('eV').magnitude, results[2])
+        else:
+            assert (
+                u_interaction,
+                u_interorbital_interaction,
+                j_hunds_coupling,
+            ) == results
+
+    @pytest.mark.parametrize(
+        'u_interaction, j_local_exchange_interaction, u_effective',
+        [
+            (3.0, 1.0, 2.0),
+            (-3.0, 1.0, -4.0),
+            (3.0, None, None),
+            (None, 1.0, None),
+        ],
+    )
+    def test_u_effective(
+        self,
+        hubbard_interactions,
+        u_interaction,
+        j_local_exchange_interaction,
+        u_effective,
+    ):
+        """
+        Test the effective Hubbard interaction `Ueff` for a given set of Hubbard interactions `U` and `J`.
+        """
+        # Adding `u_interaction` and `j_local_exchange_interaction` to the `HubbardInteractions` section
+        self.add_u_j(hubbard_interactions, u_interaction, j_local_exchange_interaction)
+
+        # Resolving Ueff from class method
+        resolved_u_effective = hubbard_interactions.resolve_u_effective(self.logger)
+        if resolved_u_effective is not None:
+            assert np.isclose(resolved_u_effective.to('eV').magnitude, u_effective)
+        else:
+            assert resolved_u_effective == u_effective
+
+    def test_normalize(self, hubbard_interactions):
+        """
+        Test the normalization of the `HubbardInteractions`. Inputs are defined as the quantities of the `HubbardInteractions` section.
+        """
+        # ? Is this enough for testing? Can we do more?
+        self.add_u_j(hubbard_interactions, 3.0, 2.0)
+        hubbard_interactions.u_interorbital_interaction = 1.0 * ureg('eV')
+        hubbard_interactions.j_hunds_coupling = 2.0 * ureg('eV')
+
+        hubbard_interactions.normalize(None, self.logger)
+        assert np.isclose(hubbard_interactions.u_effective.to('eV').magnitude, 1.0)
+        assert np.isclose(hubbard_interactions.u_interaction.to('eV').magnitude, 3.0)
+        assert hubbard_interactions.slater_integrals is None
+
+
+class TestAtomsState:
+    """
+    Tests the `AtomsState` class defined in atoms_state.py.
+    """
+
+    logger = logging.getLogger(__name__)
+
+    @staticmethod
+    def add_element_information(atom_state, quantity_name, value) -> None:
+        setattr(atom_state, quantity_name, value)
```

Comment:
not sure why you would define this one liner method, also are you if you not supposed to use m_set